src/Pretzel.Logic/Extensions/YamlExtensions.cs

```diff
@@ -73,16 +74,24 @@ private static object GetValue(YamlNode value)
             var list = value as YamlSequenceNode;
             if (list != null)
             {
-                var listResults = new List<string>();
-                foreach (var entry in list.Children)
-                {
-                    var node = entry as YamlScalarNode;
-                    if (node != null)
+                if (list.Children.All(_ => _ is YamlScalarNode)) {
+                    var listString = new List<string>();
+                    foreach (var entry in list.Children)
+                    {
+                        var node = entry as YamlScalarNode;
```

Comment:
There is: If I'd project the items with `as` and filter nulls, and compare the length of the lists. This is more readble in my opinion.